src/types.ts

```diff
@@ -102,7 +102,11 @@ export type ChangeObject<Type> = {
   [Property in keyof Type as Exclude<
     Property,
     Required<Type>[Property] extends IgnoredTypes ? Property : never
-  >]: Required<Type>[Property] extends Date ? number : Type[Property];
+  >]: Required<Type>[Property] extends Date
+    ? number
+    : Required<Type>[Property] extends Record<string | number | symbol, unknown>
+      ? string
+      : Type[Property];
```

Comment:
Diff looked crazy but basically just added one condition as below:
```
: Required<Type>[Property] extends Record<string | number | symbol, unknown>
      ? string
      : Type[Property];
```

That if it is an object, CDC turns it to false.